### ISSUE TITLE
[gui error report] Error: no protocol with name: "'dnsaddr'". Must have a valid family name

### ISSUE THREAD

**Issue by aredridel** [ipfs/ipfs-desktop#2757]
👉️ Please describe what you were doing when this error happened.

Edited my config file incorrectly, using `/dnsaddr` addresses for API and Gateway.

**Specifications**

- **OS**: darwin 23.3.0
- **IPFS Desktop Version**: 0.33.0
- **Electron Version**: 19.1.9
- **Chrome Version**: 102.0.5005.167

**Error**

```
Error: no protocol with name: "'dnsaddr'". Must have a valid family name: "{ip4, ip6, dns4, dns6}".
    at Multiaddr.nodeAddress (/Applications/IPFS Desktop.app/Contents/Resources/app.asar/node_modules/multiaddr/src/index.js:452:13)
    at checkPorts (/Applications/IPFS Desktop.app/Contents/Resources/app.asar/src/daemon/config.js:345:63)
    at getIpfsd (/Applications/IPFS Desktop.app/Contents/Resources/app.asar/src/daemon/daemon.js:71:16)
    at async startDaemon (/Applications/IPFS Desktop.app/Contents/Resources/app.asar/src/daemon/daemon.js:210:17)
    at async startIpfs (/Applications/IPFS Desktop.app/Contents/Resources/app.asar/src/daemon/index.js:49:17)
    at async setupDaemon (/Applications/IPFS Desktop.app/Contents/Resources/app.asar/src/daemon/index.js:116:3)
    at async Promise.all (index 1)
    at async run (/Applications/IPFS Desktop.app/Contents/Resources/app.asar/src/index.js:68:5)
```

There's no way in the UI to recover this.

**Comment by lidel** [comment]
You can fix this by finding `.ipfs/config` file and editing it directly, fixing the bug you introduced via GUI.

See https://github.com/ipfs/ipfs-desktop/#where-are-my-ipfs-configuration-and-log-files

If you need further support, ask https://discuss.ipfs.tech/

**Comment by aredridel** [comment]
Oh I'm well aware but uh, "I can break the UI with the UI" is maybe a bad baseline for software.

**Comment by lidel** [comment]
Thank you for explaining. Apologies for mireading your original report (it was filled as a bug, and not filled as a feature request).

I agree with the feature request: inability to edit Kubo configuration file once bad edit is done via GUI was an oversight.

Good news is that it was implemented in the [latest version](https://github.com/ipfs/ipfs-desktop/releases/latest) (v0.36). 

Now there is context menu option that allows you to find and edit the `.ipfs/config` yourself:

> ![image](https://github.com/ipfs/ipfs-desktop/assets/157609/e216fff7-4d01-451e-9d70-97a599b263a9)


If you find any problems with this, please fill new issue as this bug is marked as closed.

**Comment by aredridel** [comment]
Awesome, thank you!
